### PR TITLE
upgrade groovy from 2.5.17 to 3.0.11 to fix CVE-2019-11358(7.5)

### PR DESCRIPTION
**Pull request by hangc0276**
### Motivation
There is a CVE in testing, which was introduced by groovy.

```
Error:  Failed to execute goal org.owasp:dependency-check-maven:7.1.0:aggregate (default) on project bookkeeper: 
Error: 
Error:  One or more dependencies were identified with vulnerabilities that have a CVSS score greater than or equal to '7.0': 
Error: 
Error:  testng-6.13.1.jar: CVE-2019-[11](https://github.com/apache/bookkeeper/runs/6953376087?check_suite_focus=true#step:6:12)358(7.5)
Error: 
Error:  See the dependency-check report for more details.
Error:  -> [Help 1]
Error: 
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error: 
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Error: Process completed with exit code 1.
```

### Changes
Upgrade groovy version from 2.5.8 to 3.0.11

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1138,6 +1138,7 @@
               <mixAuditAnalyzerEnabled>false</mixAuditAnalyzerEnabled>
               <nugetconfAnalyzerEnabled>false</nugetconfAnalyzerEnabled>
               <assemblyAnalyzerEnabled>false</assemblyAnalyzerEnabled>
+              <skipSystemScope>true</skipSystemScope>
```
Comment (Shoothzj) [inline pom.xml:1141]: I have searched the project. we don't have `<scope>system<scope>` import. Why need this config?

Comment (hangc0276) [reply pom.xml:1141]: It will be failed in jdk8.
https://github.com/apache/bookkeeper/runs/6953541383?check_suite_focus=true

**Review by Shoothzj — APPROVED** [review]

**Comment by Shoothzj** [comment]
Groovy 3.0 requires JDK9+ to build and JDK8 is the minimum version of the JRE that we support. See https://groovy-lang.org/releasenotes/groovy-3.0.html
Maybe we should update to Groovy2 latest version?

**Comment by hangc0276** [comment]
> Groovy 3.0 requires JDK9+ to build and JDK8 is the minimum version of the JRE that we support. See https://groovy-lang.org/releasenotes/groovy-3.0.html Maybe we should update to Groovy2 latest version?

done

**Comment by hangc0276** [comment]
> > Groovy 3.0 requires JDK9+ to build and JDK8 is the minimum version of the JRE that we support. See https://groovy-lang.org/releasenotes/groovy-3.0.html Maybe we should update to Groovy2 latest version?
> 
> done

@Shoothzj  It doesn't work on 2.5.17, due to the groovy-testng dependent on testng 6.13.1, which has CVE issues.
https://github.com/apache/bookkeeper/runs/6954538682?check_suite_focus=true

**Comment by hangc0276** [comment]
rerun failure checks

**Review by zymap — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]

**Comment by nicoloboschi** [comment]
> Groovy 3.0 requires JDK9+ to build and JDK8 is the minimum version of the JRE that we support. See https://groovy-lang.org/releasenotes/groovy-3.0.html Maybe we should update to Groovy2 latest version?

@Shoothzj so we upgraded to groovy 3 but the JDK8 check passed, so I'm a bit confused now

**Comment by hangc0276** [comment]
> > Groovy 3.0 requires JDK9+ to build and JDK8 is the minimum version of the JRE that we support. See https://groovy-lang.org/releasenotes/groovy-3.0.html Maybe we should update to Groovy2 latest version?
> 
> @Shoothzj so we upgraded to groovy 3 but the JDK8 check passed, so I'm a bit confused now

@nicoloboschi  I have added the `<skipSystemScope>true</skipSystemScope>` configuration to skip the SystemScope ckeck.

**Pull request merged**